### PR TITLE
[boringssl] Remove -WX and -Werror

### PR DESCRIPTION
**Pull request by rpavlik**
…VS2022.

Filed upstream as https://bugs.chromium.org/p/boringssl/issues/detail?id=495

**Describe the pull request**

- #### What does your PR fix?
Build of boringssl broken as noted in https://github.com/microsoft/vcpkg/pull/24310#issuecomment-1132294821

- #### Which triplets are supported/not supported? Have you updated the [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt)?
Only affects windows. Did not update the baseline.

- #### Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?
Well, it adds a patch for a non-vcpkg-specific problem, but I did file it upstream too.

- #### If you have added/updated a port: Have you run `./vcpkg x-add-version --all` and committed the result?
yes

### CODE REVIEW

File: ports/boringssl/portfile.cmake [hunk 1]
```diff
@@ -28,6 +28,7 @@ vcpkg_from_github(
   PATCHES
     0001-vcpkg.patch
     0002-disable-waring-4065.patch
+    0003-Disable-C4191-warning-to-fix-VS2022-build.patch
 )
 
 vcpkg_configure_cmake(
```
Comment (LilyWangLL) [inline ports/boringssl/portfile.cmake:34]:
```suggestion
vcpkg_cmake_configure(
```
`vcpkg_configure_cmake` has been deprecated, and please also remove `PREFER_NINJA`. Change `vcpkg_install_cmake()` to `vcpkg_cmake_install()`

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/boringssl/portfile.cmake (hunk 1 above)
Comment (LilyWangLL) [reply ports/boringssl/portfile.cmake:34]: Also add dependency to vcpkg.json
```
"dependencies": [
    {
      "name": "vcpkg-cmake",
      "host": true
    }
]
```

**Comment by LilyWangLL** [comment]
Ping @rpavlik

**Comment by rpavlik** [comment]
Haven't had time to do the requested extra maintenance, and won't for a couple of weeks.

**Review by LilyWangLL — APPROVED** [review]

File: ports/boringssl/0003-Disable-C4191-warning-to-fix-VS2022-build.patch [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 83ff477..398e97f 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -229,6 +229,7 @@ elseif(MSVC)
+       "C5027" # move assignment operator was implicitly defined as deleted
+       "C5045" # Compiler will insert Spectre mitigation for memory load if
+               # /Qspectre switch specified
++      "C4191" # Unsafe cast (we use it as a type-erased type) - see https://bugs.chromium.org/p/boringssl/issues/detail?id=495
```
Comment (ras0219-msft) [inline ports/boringssl/0003-Disable-C4191-warning-to-fix-VS2022-build.patch:19]: Could you instead change this patch to just remove `-WX` and `-Werror` from this file? vcpkg policy is to never build with warnings-as-errors -- it just causes build breaks for customers on compiler upgrades.

**Review by ras0219-msft — CHANGES_REQUESTED** [review]
Thanks for the PR!

File: ports/boringssl/0003-Disable-C4191-warning-to-fix-VS2022-build.patch (hunk 2 above)
Comment (ras0219-msft) [reply ports/boringssl/0003-Disable-C4191-warning-to-fix-VS2022-build.patch:19]: @LilyWangLL

Comment (LilyWangLL) [reply ports/boringssl/0003-Disable-C4191-warning-to-fix-VS2022-build.patch]: Done.

**Review by ras0219-msft — APPROVED** [review]
Thanks for the PR!

**Pull request merged**